packages/web/src/common/store/smart-collection/sagas.ts

```diff
@@ -67,21 +66,9 @@ function* fetchHeavyRotation() {
     (activity: full.ActivityFull) => trackActivityFromSDK(activity)?.item
   ).filter((track) => !track.is_unlisted)
 
-  const users = yield* call(
-    retrieveUsers,
-    mostListenedTracks.map((t) => t.owner_id)
-  )
-
-  const trackIds = mostListenedTracks
-    .filter(
-      (track) =>
-        users.entries[track.owner_id] &&
-        !users.entries[track.owner_id].is_deactivated &&
-        !track.is_delete
-    )
```

Comment:
Nice! Curious how bad was the perf impact on this?
Just so I know for similar patterns in the future.

Reply:
i don't think it was too-too bad, but its at least twice as fast now :)